### PR TITLE
Adds node support, removes deprecations

### PR DESCRIPTION
**Pull request by samselikoff**

### CODE REVIEW

File: addon/index.js [hunk 1]
```diff
@@ -1,50 +1,12 @@
...
 
-import { deprecate } from '@ember/application/deprecations';
-
 Inflector.defaultRules = defaultRules;
 
-Object.defineProperty(Ember, 'Inflector', {
-  get() {
-    deprecate(`Ember.Inflector is deprecated. Please explicitly: import Inflector from 'ember-inflector';`, false, {
-      id: 'ember-inflector.globals',
-      until: '3.0.0',
```
[the remark this answers is not in the document]
Comment (stefanpenner) [reply addon/index.js]: i suspect so, but lets check with @rwjblue  and @runspired to ensure we aren't missing anything that will cause mega pain.

[the remark this answers is not in the document]
Comment (runspired) [reply addon/index.js]: I still see this deprecation printed fairly often, but I’m of the mind that at some point you have to pull the switch and give those folks a kick.

**Comment by samselikoff** [comment]
What’s the alternative - bump the deprecation to “until 4.0”, publish one
more 3.minor, then remove & publish a 4.0?

File: addon/lib/system/inflector.js [hunk 2]
```diff
@@ -1,4 +1,7 @@
-import { capitalize } from '@ember/string';
+const STRING_CAPITALIZE_REGEXP = /(^|\/)([a-z\u00C0-\u024F])/g;
+const capitalize = str => {
+  return str.replace(STRING_CAPITALIZE_REGEXP, (match /*, separator, chr */) => match.toUpperCase())
+};
```
[the remark this answers is not in the document]
Comment (rwjblue) [reply addon/lib/system/inflector.js]: Agree with @stefanpenner here, but FWIW we are (slowly) working on extracing `@ember/string` to its own stand alone repo / npm package: https://github.com/emberjs/ember-string (for [this RFC](https://github.com/emberjs/rfcs/blob/master/text/0236-deprecation-ember-string.md)).

File: node-tests/.eslintrc.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
+module.exports = {
```
Comment (rwjblue) [inline node-tests/.eslintrc.js]: I'd rather use `overrides` in the the top level `.eslintrc.js` than mkae a new file here...

File: package.json [hunk 4]
```diff
@@ -48,9 +49,13 @@
     "ember-try": "^0.2.23",
     "eslint-plugin-ember": "^5.0.0",
     "eslint-plugin-node": "^6.0.1",
-    "loader.js": "^4.2.3"
+    "esm": "^3.0.63",
+    "loader.js": "^4.2.3",
+    "qunit": "^2.6.1"
   },
+  "main": "addon/index.js",
```
Comment (rwjblue) [inline package.json]: Hmm, I think you need to follow the normal esm suggested path for this to work.

* add a `main.js` that does:

```js
// Set options as a parameter, environment variable, or rc file.
require = require("esm")(module/*, options*/);
module.exports = require("./addon");
```

* move `esm` to a `dependency`
* change tests to not use `esm` directly

File: addon/index.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (rwjblue) [reply addon/index.js]: I don't think we can delete them in the current major version, but I'd be in favor of dropping them and bumping to 4.0.0...

**Comment by btecu** [comment]
This can be closed.

**Pull request closed without merging**